volttron/platform/dbutils/basedb.py

```diff
@@ -96,9 +96,19 @@ def __init__(self, dbapimodule, **kwargs):
         self.__connect = connect
         self.__connection = None
         self.stash = local()
+        self.max_topic_id = None
+
+    def get_max_topic_id(self):
+        """
+        Function to return max topic id in the database. This can be used to return last inserted topic_id so that
+        concrete classes can generate topic_ids locally and bulk insert topics. Base class returns None
+        :returns: max topic id in database
+        :rtype: int
+        """
+        return None
 
     @contextlib.contextmanager
-    def bulk_insert(self):
+    def bulk_insert_data(self):
```

Comment:
We need to support bulk_insert as an abstract method.  Any historian (we don't control) will break with this change.

Reply:
Agreed. Renamed bulk_insert_data back to bulk_insert